sigstore/_internal/trust.py

```diff
@@ -575,6 +536,65 @@ def from_json(cls, raw: str) -> ClientTrustConfig:
         inner = _ClientTrustConfig().from_json(raw)
         return cls(inner)
 
+    @classmethod
+    def production(
+        cls,
+        offline: bool = False,
+    ) -> ClientTrustConfig:
+        """Create new trust config from Sigstore production TUF repository.
+
+        If `offline`, will use data in local TUF cache. Otherwise will
+        update the data from remote TUF repository.
+        """
+        return cls.from_tuf(DEFAULT_TUF_URL, offline)
+
+    @classmethod
+    def staging(
+        cls,
+        offline: bool = False,
+    ) -> ClientTrustConfig:
+        """Create new trust config from Sigstore staging TUF repository.
+
+        If `offline`, will use data in local TUF cache. Otherwise will
+        update the data from remote TUF repository.
+        """
+        return cls.from_tuf(STAGING_TUF_URL, offline)
+
+    @classmethod
+    def from_tuf(
+        cls,
+        url: str,
+        offline: bool = False,
+    ) -> ClientTrustConfig:
+        """Create a new trust config from a TUF repository.
+
+        If `offline`, will use data in local TUF cache. Otherwise will
+        update the trust config from remote TUF repository.
+        """
+        updater = TrustUpdater(url, offline)
+
+        tr_path = updater.get_trusted_root_path()
+        inner_tr = _TrustedRoot().from_json(Path(tr_path).read_bytes())
+
+        try:
+            sc_path = updater.get_signing_config_path()
+            inner_sc = _SigningConfig().from_json(Path(sc_path).read_bytes())
+        except TUFError as e:
+            # TUF repo may not have signing config yet: hard code values for prod:
+            if url == DEFAULT_TUF_URL:
+                embedded = read_embedded("signing_config.v0.2.json", url)
+                inner_sc = _SigningConfig().from_json(embedded)
+            else:
+                raise e
```

Comment:
Just confirming my understanding: we're doing this because the production TUF instance doesn't have the v2 signing config yet, right? So we'll be able to remove this once it's deployed there?

Reply:
(And if that's right, is there a tracking issue/PR we can xref here?)

Reply:
correct. I will file an issue